File: python/query_points_test.py
from OceanDB.AlongTrack import AlongTrack
import matplotlib.pyplot as plt
import numpy as np
import datetime

atdb = AlongTrack(db_name='ocean')

# data = atdb.geographic_points_in_spatialtemporal_window(11.375, -79.17, 500000, '2002-08-01', '2002-09-01')
# data = atdb.geographic_points_in_spatialtemporal_window(11., -80, 800000, '1992-05-15', '1994-05-25', should_basin_mask=1)
# data = atdb.geographic_points_in_spatialtemporal_window(11, -80, 800000, '2002-05-15', '2002-05-25', should_basin_mask=0)
# data = atdb.geographic_points_in_spatialtemporal_window(11, 150, 800000, '2002-05-15', '2002-05-25', should_basin_mask=0)

data = atdb.geographic_points_in_spatialtemporal_window(11, 150, datetime.datetime(2022,5,15))

aa = np.array(data)

x = np.array([data_i[0] for data_i in data])
y = np.array([data_i[1] for data_i in data])
sla = np.array([data_i[2] for data_i in data])
t = np.array([data_i[3] for data_i in data])

plt.figure()
plt.scatter(x, y, c=sla)
plt.show()


[x, y, sla, t] = atdb.projected_points_in_spatialtemporal_window(11, 150, datetime.datetime(2022,5,15))
plt.figure()
plt.scatter(x, y, c=sla)
plt.show()

# print(data)

File: matlab/query_points_test.m
atdb = AlongTrack();

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%
% Geographic Points
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

results = atdb.geographicPointsInSpatialtemporalWindow(11, -80, datetime(2002,5,15),distance=1000e3);

figure(Name="Geographic points")
scatter(results.longitude,results.latitude,5^2,results.sla_filtered,"filled");

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%
% Projected Points
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Lx=2000e3;
Ly=2000e3;

results = atdb.projectedPointsInSpatialtemporalWindow(50, -138, datetime(2002,5,15),Lx=Lx,Ly=Ly);

% results = atdb.projectedPointsInSpatialtemporalWindow(11, -80, datetime(2002,5,15),Lx=Lx,Ly=Ly);

figure(Name="Projected points")
tiledlayout(2,1)
nexttile
scatter(results.longitude,results.latitude,5^2,results.sla_filtered,"filled");
nexttile
scatter(results.x/1e3,results.y/1e3,5^2,results.sla_filtered,"filled");
axis equal
xlim([0 Lx/1e3])
ylim([0 Ly/1e3])

d = sqrt((results.x-Lx/2).^2 + (results.y-Ly/2).^2);
figure, scatter(results.distance,results.distance-d,'filled')